dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -193,13 +196,23 @@ class AsyncRequestHandler : public UnaryCallback<bool>
     {
         switch (stage)
         {
+        case AsyncRequestStage::WAIT_RETRY:
+            start();
+            break;
         case AsyncRequestStage::WAIT_MAKE_READER:
         {
             // Use lock to ensure reader is created already in reactor thread
             std::unique_lock lock(mu);
             if (!ok)
+            {
                 reader.reset();
-            notifyReactor();
+                LOG_WARNING(log, "MakeReader fail. retry time: {}", retry_times);
+                retryOrDone("Exchange receiver meet error : send async stream request fail");
```

Comment:
looks like you need to notify the reactor thread if there is no more try?

Reply:
Done